dipy/direction/peaks.py

```diff
@@ -572,3 +571,49 @@ def reshape_peaks_for_visualization(peaks):
         peaks = peaks.peak_dirs
 
     return peaks.reshape(np.append(peaks.shape[:-2], -1)).astype('float32')
+
+
+def peaks_from_positions(positions, odfs, sphere, relative_peak_threshold=.5,
+                         min_separation_angle=25, is_symmetric=True, npeaks=5):
+    """
+    Extract the peaks at each positions.
+
+    Parameters
+    ----------
+    position : array, (N, 3)
+        Voxel coordinates of the N positions.
+    odfs : array, (X, Y, Z, M)
+        Orientation distribution function (sphericla function) represented
+        on a sphere of M points.
+    sphere : Sphere
+        A discrete Sphere. The M points on the sphere correspond to the points
+        of the odfs.
+    relative_peak_threshold : float in [0., 1.]
+        Only peaks greater than ``min + relative_peak_threshold * scale`` are
+        kept, where ``min = max(0, odf.min())`` and
+        ``scale = odf.max() - min``.
+    min_separation_angle : float in [0, 90]
+        The minimum distance between directions. If two peaks are too close
+        only the larger of the two is returned.
+    is_symmetric : bool, optional
+        If True, v is considered equal to -v.
+    npeaks : int
+        The maximum number of peaks to extract at from each position.
+
+    Returns
+    -------
+    peaks_arr : array (N, npeaks, 3)
+    """
+    if positions.dtype is not np.float64:
+        positions = positions.astype(np.float64)
+
+    peaks_arr = np.zeros((len(positions), npeaks, 3))
+
+    for i, s in enumerate(positions):
+        odf = trilinear_interpolate4d(odfs, s)
+        peaks, _, _ = peak_directions(odf, sphere, relative_peak_threshold,
+                                      min_separation_angle, is_symmetric)
+        n = min(npeaks, peaks.shape[0])
```

Comment:
can you avoid one letter variable (`n`).  3 letters is the minimal

Reply:
yes, fixed.